tests/run-tests.sh

```diff
@@ -26,18 +26,19 @@ export QUARTO_DEBUG=true
 QUARTO_DENO_OPTIONS="--config test-conf.json --unstable --allow-read --allow-write --allow-run --allow-env --allow-net --check"
 
 
-# Ensure that we've restored the renv
-Rscript -e "if (!requireNamespace('renv', quietly = TRUE)) install.packages('renv')"
-Rscript -e "renv::restore()"
-
-# Ensure that we've actived the python env
-if [[ $QUARTO_TESTS_VIRTUALENV != "FALSE" ]]; then
-  source bin/activate
-  python3 -m pip install -r requirements.txt --prefer-binary -q
+if [[ -z $GITHUB_ACTION ]] && [[ -z $QUARTO_TESTS_NO_CONFIG ]]
+then
+  echo "> Checking and configuring environment for tests"
+  source configure-test-env.sh
 fi
 
-# Ensure that tinytex is installed
-quarto install tinytex --no-prompt
+# Activating python virtualenv
+if [[ -z $VIRTUAL_ENV ]]
```

Comment:
```
if [[ -z $FORCE_NO_PIPENV ]]
then
  OLD_VIRTUAL_ENV=$VIRTUAL_ENV
  if [[ -n $VIRTUAL_ENV ]]
  then
    deactivate
  end
  echo "> Activating virtualenv for Python tests"
  source "$(pipenv --venv)/bin/activate"
  quarto_venv_activated="true"
end
# ...
# at the end
# this is probably not correct if we were not in a venv but you get the idea.
if [[ -n $OLD_VIRTUAL_ENV ]]
then
  deactivate
  source $OLD_VIRTUAL_ENV/bin/activate
end

```

Reply:
So I did another approach as deactivating was not robust. For example I am using `pyenv` on Linux to manage my python and virtualenv  and `deactivate` was not existing. 

So I leveraged some **pipenv** feature. Here is how it works

* `pipenv` will always create a virtual environment even when run in another virtuaenv. This means by default it will activate the new virtualenv. This environment will be located in  `tests/.venv`
* If I detect that $VIRTUAL_ENV is set though before activating with pipenv, I remember this, and at the end, I run `source $OLD_VIRTUAL_ENV/bin/activate` to reactivate the initial virtualenvironment. 
* To skip the default behavior `QUARTO_TESTS_FORCE_NO_PIPENV` can be set to any value. In that case, tests are running in the current python environment with no change. 

I believe this work well and answer our discussion.